### PR TITLE
Updates to support Swift 6 language version

### PR DESCRIPTION
**Pull request by natecook1000**
Some minor NFC changes to resolve strict concurrency checking warnings/errors and the warning about the change in meaning for `#file`.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

File: Plugins/GenerateManual/GenerateManualPluginError.swift [hunk 1]
```diff
@@ -10,7 +10,11 @@
 //===----------------------------------------------------------------------===//
 
 import Foundation
+#if swift(>=6)
+@preconcurrency import PackagePlugin
```
Comment (rauhul) [inline Plugins/GenerateManual/GenerateManualPluginError.swift:14]: Hmm it feels like we should file a bug on SwiftPM for this

**Review by rauhul — APPROVED** [review]
I dont understand the file -> filePath change but looks right overall

File: Plugins/GenerateManual/GenerateManualPluginError.swift (hunk 1 above)
Comment (natecook1000) [reply Plugins/GenerateManual/GenerateManualPluginError.swift:14]: I can't reproduce the warnings that led me to add this anymore 😭

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**